pkg/util/util.go

```diff
@@ -179,3 +180,36 @@ func CheckWithRegex(pattern, name string) bool {
 
 	return reg.MatchString(name)
 }
+
+func GetMappedGitOpsComponent(component appstudiov1alpha1.Component) gitopsgenv1alpha1.Component {
+	gitopsMapComponent := &gitopsgenv1alpha1.Component{}
+
+	if component.Spec.Source.ComponentSourceUnion.GitSource != nil {
+		gitopsMapComponent = &gitopsgenv1alpha1.Component{
+			Spec: gitopsgenv1alpha1.ComponentSpec{
+				ComponentName: component.Spec.ComponentName,
+				Application:   component.Spec.Application,
+				Secret:        component.Spec.Secret,
+				Source: gitopsgenv1alpha1.ComponentSource{
+					ComponentSourceUnion: gitopsgenv1alpha1.ComponentSourceUnion{
+						GitSource: &gitopsgenv1alpha1.GitSource{
+							URL:           component.Spec.Source.ComponentSourceUnion.GitSource.URL,
```

Comment:
`component.Spec.Source.ComponentSourceUnion.GitSource` is a pointer; so you will need to check if its `nil` before using it

Reply:
Also new function means new tests to keep the test coverage up 😄

Reply:
Tests added